dao/video-seek.go

```diff
@@ -0,0 +1,52 @@
+package dao
+
+import (
+	"github.com/joschahenningsen/TUM-Live/model"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+const maxChunksPerVideo = 150
+
+type VideoSeekDao interface {
+	Add(streamID string, pos float64) error
+	Get(streamID string) ([]model.VideoSeekChunk, error)
+}
+
+type videoSeekDao struct {
+	db *gorm.DB
+}
+
+func NewVideoSeekDao() VideoSeekDao {
+	return videoSeekDao{db: DB}
+}
+
+func (d videoSeekDao) Add(streamID string, pos float64) error {
+	var stream *model.Stream
+	if err := DB.First(&stream, "id = ?", streamID).Error; err != nil {
+		return err
+	}
+
+	hitPos := float64(stream.Duration) * pos
```

Comment:
In typescript `pos` is calculated by `player.currentTime() / player.duration()` and this value is then multiplied with `stream.Duration` in `.Add(...)`. 

Shouldn't this be the same as `player.currentTime()`? Or is the problem that `stream.Duration != player.duration()`?

Reply:
Yes that would be much more straight forward! I think the duration should be the same 👍